### PR TITLE
build: use pre-commits to lint the lib

### LINKED ISSUE #191: use the black formatter ?
same question as in ipyvue: 

> AS you mentioned that using black would be a good idea (#40 (comment)). would you like to apply it on the whole lib ?
> I could set up some GitHub actions (including test + formatting) at the same time.

### PR DESCRIPTION
**Pull request by 12rambau**
I added a .pre-commit config file to lint the lib using the following tools:

- black
- flake8
- prettier
- commitizen
- isort

I also took the liberty to add 2 badges, and some configuration in the `setup.cfg`

Fix #191

### CODE REVIEW

**Comment by maartenbreddels** [comment]
Nice, I’m in favour of this, but maybe replace isort and black with ruff?

**Comment by 12rambau** [comment]
I wasn't aware of ruff, let me update the config and run it again

**Comment by 12rambau** [comment]
@maartenbreddels on the ruff readme they suggest to replace `flake8` and `isort` but not `black`:
> As a project, Ruff is designed to be used alongside Black and, as such, will defer implementing stylistic lint rules that are obviated by autoformatting.

 should I go in this direction ?

**Comment by maartenbreddels** [comment]
Let’s see what Mario thinks

**Comment by mariobuikhuizen** [comment]
Looks good to me! Thanks!

Regarding ruff, I don't know much about it, but if the authors recommend not replacing black, I'd go with that.

File: .gitignore [hunk 1]
```diff
@@ -16,6 +16,9 @@ ipyvuetify/labextension/
 ipyvuetify/nbextension/
 js/lib
 
-#readthedocs
+# readthedocs
 _build/
 
+# vs code 
+.vscode/
```
Comment (mariobuikhuizen) [inline .gitignore:23]: This is not related to using pre-commit. Also, we want to limit this to files specific to the project,  not to the environment of the developer. The global .gitignore can be used for this (in [home]/.gitignore).

**Review by mariobuikhuizen — CHANGES_REQUESTED** [review]
Thanks for this contribution @12rambau!

Could you remove the package-lock files? (the changes are probably caused by using a different version of nodejs)

After that, can you squash the commits? I think this change can be one commit.

File: .gitignore (hunk 1 above)
Comment (12rambau) [reply .gitignore:23]: that's funny I had the exact same argument few weeks ago in the commitizen repository https://github.com/commitizen-tools/commitizen/pull/657 but they were using a broader .gitignore file so I think I can't win here ;-)

**Comment by 12rambau** [comment]
I rollback to the previous version of the package-lock files and removed ".vscode" from the `.gitignore`. 

I didn't squash anything from my side as I don't think the intermediate commits are meaningful. Could you instead squash it directly using the merge button. side advantage, you'll be able to select the commit message (I'm never sure if I'm using the appropriate conventional commit classifier)

PS: please do that for any of my PR (we are doing the same in pydata-sphinx-theme and that gives nice changelog and easy to follow commit history).

File: .gitignore (hunk 1 above)
Comment (mariobuikhuizen) [reply .gitignore]: I see you won there :-). It's indeed a preference, but don't you think having `.vscode` (or `.idea` in my case) in `~/.gitignore` is a good solution for developer specific files? The developer sets this up once for their system, then it works for all projects.

**Comment by mariobuikhuizen** [comment]
Yeah, I can do that 👍 , but I thought you might want to write the commit message and have the commit on your name.

**Review by mariobuikhuizen — APPROVED** [review]

**Comment by 12rambau** [comment]
the commit will still be on my  name no worries
